### PR TITLE
[Notifier] Add Esendex message ID to SentMessage object

### PR DESCRIPTION
**Pull request by benr77**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

The Esendex API request returns a unique message ID for each SMS sent.

This commit simply extracts the message ID and adds it to the returned SentMessage object.

### CODE REVIEW

File: src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php [hunk 1]
```diff
@@ -69,6 +69,9 @@ protected function doSend(MessageInterface $message): SentMessage
 
         $response = $this->client->request('POST', 'https://'.$this->getEndpoint().'/v1.0/messagedispatcher', [
             'auth_basic' => sprintf('%s:%s', $this->email, $this->password),
+            'headers' => [
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php:72]: Why is this needed?

AFAIK the header is set automatically when using json key 🤔

cc @nicolas-grekas

File: src/Symfony/Component/Notifier/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ CHANGELOG
 ---
 
  * Add `SentMessageEvent` and `FailedMessageEvent`
+ * Add returned message ID to `SentMessage` in `EsendexTransport`
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/CHANGELOG.md]: This should be added to the changelog of the bridge, not the notifier component

File: src/Symfony/Component/Notifier/Bridge/Esendex/Tests/EsendexTransportTest.php [hunk 3]
```diff
@@ -88,4 +89,26 @@ public function testSendWithErrorResponseContainingDetailsThrowsTransportExcepti
...
+
+        $client = new MockHttpClient(static function () use ($response): ResponseInterface {
+            return $response;
+        });
+
+        $transport = $this->createTransport($client);
+
+        $sentMessage = $transport->send(new SmsMessage('phone', 'testMessage'));
+
+        $this->assertEquals($messageId, $sentMessage->getMessageId());
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Esendex/Tests/EsendexTransportTest.php]: assertSame please

File: src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php [hunk 4]
```diff
@@ -82,7 +85,11 @@ protected function doSend(MessageInterface $message): SentMessage
         }
 
         if (200 === $statusCode) {
-            return new SentMessage($message, (string) $this);
+            $result = $response->toArray();
+            $sentMessage = new SentMessage($message, (string) $this);
+            $sentMessage->setMessageId($result['batch']['messageheaders'][0]['id']);
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php]: We should only set this if all keys exists, can you please add a check? Thanks

File: src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php (hunk 1 above)
Comment (benr77) [reply src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php:72]: I found that Esendex returned XML by default unless I added this - but it was tested on 5.3 - have the headers being set automatically been added in 5.4?

File: src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php (hunk 4 above)
Comment (benr77) [reply src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php]: Do you want me to check it like this?

`if (isset($result['batch']) && isset($result['batch']['messageheaders']) && isset($result['batch']['messageheaders'][0]) && isset($result['batch']['messageheaders'][0]['id']))`

Comment (vuphuong87) [reply src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php]: @benr77 something liike this?

```
$messageId = $result['batch']['messageheaders'][0]['id'] ?? null;
if ($messageId) {
   $sentMessage->setMessageId($messageId);
}
```

File: src/Symfony/Component/Notifier/Bridge/Esendex/CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,8 @@
 CHANGELOG
 =========
 
+ * Add returned message ID to `SentMessage`
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Esendex/CHANGELOG.md:4]:
```suggestion
5.4
---

 * Add returned message ID to `SentMessage`
```

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php (hunk 1 above)
Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/Esendex/EsendexTransport.php:72]: Oh looks like it only sets Content-Type header

image

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @benr77.
